libfuse/file.go

```diff
@@ -11,19 +11,41 @@ import (
 	"golang.org/x/net/context"
 )
 
+type eiCache struct {
+	ei  libkbfs.EntryInfo
+	rid string
+}
+
 // File represents KBFS files.
 type File struct {
 	folder *Folder
 	node   libkbfs.Node
+
+	// for the Attr call after Create
+	eiCache *eiCache
 }
 
 var _ fs.Node = (*File)(nil)
 
+func fillAttrWithMode(ei *libkbfs.EntryInfo, a *fuse.Attr) {
+	fillAttr(ei, a)
+	a.Mode = 0644
+	if ei.Type == libkbfs.Exec {
+		a.Mode |= 0111
+	}
+}
+
 // Attr implements the fs.Node interface for File.
 func (f *File) Attr(ctx context.Context, a *fuse.Attr) (err error) {
 	f.folder.fs.log.CDebugf(ctx, "File Attr")
 	defer func() { f.folder.reportErr(ctx, libkbfs.ReadMode, err) }()
 
+	if f.eiCache != nil && ctx.Value(CtxIDKey) == f.eiCache.rid {
+		fillAttrWithMode(&f.eiCache.ei, a)
+		f.eiCache = nil
```

Comment:
Yeah I'll add a comment. Although as a pre-caution in case bazil changes, I guess we could use `atomic` for loading/storing `f.eiCache`. It wouldn't make it race-free (although I think the race is fine here since it's just setting `f.eiCache` to nil, which is not conflicting with presumably race situation where f.eiCache is already nil before line 45) but it'd eliminate the possibility of dereferencing null-pointer `f.ei.Cache` here